test/cross-browser/reference.html

```diff
@@ -22,36 +20,51 @@
 
     <!-- traces -->
     <script>
-        var childWin;
+        var clientLogger;
+
+        function createLocalStorageLogger() {
+            return {
+                debug: function() {
+                    localStorage.setItem(
+                        String(Date.now()),
+                        arguments['0'] + ' ' + arguments['1'] + ' ' + arguments['2']
+                    );
+                }
+            }
+        }
 
         var createRcSocket = function() {
-            var child = window.open('', 'childLogger');
-            childWin = child.window;
-
-            if (!childWin.clientLogger) {
-                childWin.clientLogger = {
-                    buffer: [],
-
-                    debug: function() {
-                        childWin.clientLogger.buffer.push({
-                            ts: Date.now(),
-                            message: arguments['0'] + ' ' + arguments['1'] + ' ' + arguments['2']
-                        });
-                    }
-                };
+            if(typeof(Storage) !== "undefined") {
+                clientLogger = createLocalStorageLogger();
+            } else {
+                console.log("no Web Storage support");
             }
 
             var rc = new RcSocket("ws://localhost:9998/");
-            rc.logger = childWin.clientLogger.debug;
+            rc.logger = clientLogger.debug;
             rc.debug = true;
+
             return rc;
         };
 
+        var dumpLogger = function() {
+            var buffer = [];
+            for (var i = 0; i < localStorage.length; i++) {
+                buffer.push({
+                    ts: parseInt(localStorage.key(i)),
+                    message: localStorage.getItem(localStorage.key(i))
+                });
+            }
```

Comment:
I lightly tested this inside of a chrome console and it worked. Can't speak for other browsers.

```
for (var key in localStorage) {
}
```